### DISCUSSION TITLE
Silero without pytorch dep

### DISCUSSION THREAD

**Discussion by roablep** [snakers4/silero-vad#569] [Q&A] [answered]
Is it possible to use Silero without the pytorch dependency and only onnx?
If not, is it possible to use a lighter version of pytorch like cpuonly?

**Comment by snakers4** [comment, marked as answer]
> Is it possible to use Silero without the pytorch dependency and only onnx?

Yes it is, but you will need to go through the code and delete all the PyTorch references and just load the onnx model from local file. The onnx version can work standalone, but you will also need to replace torchaudio with your audio loading library.

> If not, is it possible to use a lighter version of pytorch like cpuonly?

As far as I know, there CPU only versions of PyTorch listed on the installation page. Also if you can use c++, you can use libtorch.
